src/hotspot/cpu/x86/register_x86.hpp

```diff
@@ -257,9 +257,9 @@ class ConcreteRegisterImpl : public AbstractRegisterImpl {
   // it's optoregs.
 
     number_of_registers = RegisterImpl::number_of_registers * RegisterImpl::max_slots_per_register +
-      2 * FloatRegisterImpl::number_of_registers +
+      2 * FloatRegisterImpl::number_of_registers + NOT_LP64(8) LP64_ONLY(0) +
```

Comment:
This is to cover for additional FILL0-7 definitions present only for 32 bit JVM

Reply:
Hm, still not clear to me why it matters only now. 
Is it because predicate registers are put at the very end and the wrong sizing leaves them out of the RegMask bit map on x86-32?

Reply:
Yes , its needed for correct computation RM_SIZE (number of of words to hold a RegMask), till now we were not accounting FILL register count while computing total number of registers, but it was not hurting since number of words is the ceiling multiple of word size.